amqphandler/amqphandler.go

```diff
@@ -282,6 +295,14 @@ func (handler *AmqpHandler) SendOverrideEnvVarsStatus(envs cloudprotocol.Overrid
 	return nil
 }
 
+// SubscribeForConnectionEvents subscribes for connection events.
+func (handler *AmqpHandler) SubscribeForConnectionEvents(consumer ConnectionEventsConsumer) {
```

Comment:
There should be an unsubscribe method to avoid notifications of the closed objects, or we have to follow strict rules of the closed consumers before closing the sender.

Reply:
Unsubscribe method added.